CMakeLists.txt

```diff
@@ -25,6 +25,7 @@ set(PROJECT_VERSION "2.3.0")
 # Avoid SYCL compiler error
 if(NOT WIN32)
   string(APPEND CMAKE_CXX_FLAGS " -Wno-error")
+  string(APPEND CMAKE_CXX_FLAGS " -Werror=unused-variable")
```

Comment:
Enabling this option is good for the final code quality, but it also requires every compilation in development to follow this rule. Such checks would be better placed in CI.

Reply:
Ok, I've passed this flag as env variable in linux_build CI pipeline